Makefile: race condition between install-schemas and compile-schemas

**Describe the bug**

The Makefile contains an [`install-system` target](https://github.com/Guake/guake/blob/47312dc2481b2fa617afe79cdcfddadf60e84516/Makefile#L71) that depends on both `install-schemas` and `compile-shemas` (sic).

Hoever: the `compile-shemas` target reads files from paths that are created by the `install-schemas` target, and they can both run in parallel when `make`'s parallelism is enabled (`-j`).

**Expected behavior**

The `.gschema.xml` files required for the `compile-shemas` target should always exist before the `compile-shemas` target is run.

**Actual behavior**

When `make` parallelism is enabled a `gschemas.compiled` file may or may not be created, depending on the runtime behaviour of the race condition.

**To Reproduce**

The sample below uses `make -j12 install-system` to approximate the behaviour:

```sh
$ PREFIX=/.../bugreport/ make -j12 install-system
install -dm755                                       "//.../bugreport//share/applications"
if [ 1 = 1 ]; then glib-compile-schemas //.../bugreport//share/glib-2.0/schemas; fi
for f in $(find po -iname "*.mo"); do \
        l="${f%%.*}"; \
        lb=$(basename $l); \
        install -Dm644 "$f" "//.../bugreport//share/locale/$lb/LC_MESSAGES/guake.mo"; \
done;
install -Dm644 "/.../guake/data/guake.desktop"       "//.../bugreport//share/applications/"
# you probably want to execute this target with sudo:
Error opening directory “//.../bugreport//share/glib-2.0/schemas”: No such file or directory
make: *** [Makefile:147: compile-shemas] Error 1
```

Note: sometimes the output differs, due to the race condition; for example if the directory has been created but does not yet contain `.gschema.xml` files then a message `No schema files found: doing nothing.` appears instead.

--- a/Makefile
+++ b/Makefile
@@ -126,7 +126,6 @@ uninstall-dev-locale:
 	@rm -rf guake/po
 
 install-schemas:
-	if [ $(COMPILE_SCHEMA) = 1 ]; then glib-compile-schemas $(DESTDIR)$(SCHEMA_DIR); fi
 	install -dm755                                       "$(DESTDIR)$(datadir)/applications"
 	install -Dm644 "$(DEV_DATA_DIR)/guake.desktop"       "$(DESTDIR)$(datadir)/applications/"
 	install -Dm644 "$(DEV_DATA_DIR)/guake-prefs.desktop" "$(DESTDIR)$(datadir)/applications/"
@@ -143,6 +142,7 @@ install-schemas:
 	install -Dm644 "$(DEV_DATA_DIR)"/*.glade "$(DESTDIR)$(GLADE_DIR)/"
 	install -dm755                                         "$(DESTDIR)$(SCHEMA_DIR)"
 	install -Dm644 "$(DEV_DATA_DIR)/org.guake.gschema.xml" "$(DESTDIR)$(SCHEMA_DIR)/"
+	if [ $(COMPILE_SCHEMA) = 1 ]; then glib-compile-schemas $(DESTDIR)$(SCHEMA_DIR); fi
 
 uninstall-system: uninstall-schemas uninstall-locale
 	$(SHELL) -c $(PYTHON_SITEDIRS_FOR_PREFIX) \

--- a/Makefile
+++ b/Makefile
@@ -68,7 +68,7 @@ ln-venv:
 clean-ln-venv:
 	@rm -f .venv
 
-install-system: install-schemas compile-shemas install-locale install-guake
+install-system: install-schemas install-locale install-guake
 
 install-guake:
 	# you probably want to execute this target with sudo:
@@ -126,6 +126,7 @@ uninstall-dev-locale:
 	@rm -rf guake/po
 
 install-schemas:
+	if [ $(COMPILE_SCHEMA) = 1 ]; then glib-compile-schemas $(DESTDIR)$(SCHEMA_DIR); fi
 	install -dm755                                       "$(DESTDIR)$(datadir)/applications"
 	install -Dm644 "$(DEV_DATA_DIR)/guake.desktop"       "$(DESTDIR)$(datadir)/applications/"
 	install -Dm644 "$(DEV_DATA_DIR)/guake-prefs.desktop" "$(DESTDIR)$(datadir)/applications/"
@@ -142,9 +143,6 @@ install-schemas:
 	install -Dm644 "$(DEV_DATA_DIR)"/*.glade "$(DESTDIR)$(GLADE_DIR)/"
 	install -dm755                                         "$(DESTDIR)$(SCHEMA_DIR)"
 	install -Dm644 "$(DEV_DATA_DIR)/org.guake.gschema.xml" "$(DESTDIR)$(SCHEMA_DIR)/"
-
-compile-shemas:
-	if [ $(COMPILE_SCHEMA) = 1 ]; then glib-compile-schemas $(DESTDIR)$(gsettingsschemadir); fi
 
 uninstall-system: uninstall-schemas uninstall-locale
 	$(SHELL) -c $(PYTHON_SITEDIRS_FOR_PREFIX) \

--- a/releasenotes/notes/schema-compilation-race-condition-fix-4e637a29a86126ae.yaml
+++ b/releasenotes/notes/schema-compilation-race-condition-fix-4e637a29a86126ae.yaml
@@ -2,4 +2,5 @@ release_summary: >
   Fix a build race condition between compilation/installation of GSettings schemas
 
 fixes:
-  - Ensure that guake's GSettings XML schemas are installed before requesting their (re)compilation #2219
+  - |
+      - Ensure that guake's GSettings XML schemas are installed before requesting their (re)compilation #2219

--- a/releasenotes/notes/schema-compilation-race-condition-fix-4e637a29a86126ae.yaml
+++ b/releasenotes/notes/schema-compilation-race-condition-fix-4e637a29a86126ae.yaml
@@ -1,0 +1,5 @@
+release_summary: >
+  Fix a build race condition between compilation/installation of GSettings schemas
+
+fixes:
+  - Ensure that guake's GSettings XML schemas are installed before requesting their (re)compilation #2219

